packages/medusa/src/strategies/batch-jobs/product/import.ts

```diff
@@ -0,0 +1,732 @@
+
+    const variantOps = await this.downloadImportOpsFile(
+      batchJobId,
+      OperationType.VariantCreate
+    )
+
+    for (const variantOp of variantOps) {
+      try {
+        const variant = transformVariantData(variantOp)
+
+        const product = await this.productService_
+          .withTransaction(transactionManager)
+          .retrieveByHandle(variantOp["product.handle"] as string, {
+            relations: ["variants", "variants.options", "options"],
+          })
+
+        const optionIds =
+          (variantOp["product.options"] as Record<string, string>[])?.map(
+            (variantOption) =>
+              product!.options.find(
+                (createdProductOption) =>
+                  createdProductOption.title === variantOption.title
+              )!.id
+          ) || []
+
+        variant.options =
+          (variant.options as Record<string, any>[])?.map((o, index) => ({
+            ...o,
+            option_id: optionIds[index],
+          })) || []
+
+        await this.productVariantService_
+          .withTransaction(transactionManager)
+          .create(product!, variant as unknown as CreateProductVariantInput)
+
+        this.updateProgress(batchJobId)
+      } catch (e) {
+        ProductImportStrategy.throwDescriptiveError(variantOp)
+      }
+    }
+  }
+
+  /**
+   * Method updates product variants from a CSV data.
+   *
+   * @param batchJobId - An id of the current batch job being processed.
+   */
+  private async updateVariants(batchJobId: string): Promise<void> {
+    const transactionManager = this.transactionManager_ ?? this.manager_
+    const productOptionRepo = this.manager_.getCustomRepository(
+      this.productOptionRepo_
+    )
+
+    const variantOps = await this.downloadImportOpsFile(
+      batchJobId,
+      OperationType.VariantUpdate
+    )
+
+    for (const variantOp of variantOps) {
+      try {
+        await this.prepareVariantOptions(variantOp, productOptionRepo)
+
+        await this.productVariantService_
+          .withTransaction(transactionManager)
+          .update(
+            variantOp["variant.id"] as string,
+            transformVariantData(variantOp) as UpdateProductVariantInput
+          )
+      } catch (e) {
+        ProductImportStrategy.throwDescriptiveError(variantOp)
+      }
+
+      this.updateProgress(batchJobId)
+    }
+  }
+
+  /**
+   * Extend records used for creating variant options with corresponding product option ids.
+   *
+   * @param variantOp - Parsed row data form CSV
+   * @param productOptionRepo - ProductOption repository
+   */
+  protected async prepareVariantOptions(
+    variantOp,
+    productOptionRepo: ProductOptionRepository
+  ): Promise<void> {
+    const productOptions = variantOp["variant.options"] || []
+
+    for (const o of productOptions) {
+      const { id } = (await productOptionRepo.findOne({
+        where: { title: o._title },
+      })) as ProductOption
```

Comment:
todo: this is insufficient as we would have to get the correct product option for the product being worked on. For example, this could exist in the DB:

- Product Option 1: `{ id: "option_1", title: "Size", product_id: "prod_a" }`
- Product Option 2: `{ id: "option_2", title: "Size", product_id: "prod_b" }`

Now let's say we are adding a variant to Product B; we could end up having a situation where the variant has a product option value like this:

- Product Option Value: { id: "optval_1", value: "Medium", **option_id: "option_1"**, variant_id: "new_variant" }

In this case, the option value refers to a ProductVariant belonging to **Product B** but a ProductOption belonging to **Product A**.

In the case where we are working with an existing product option we should be able to structure the query to be something along the lines of:
```
{ title: o._title, product_id: variantOp.product_id },
```